### PR TITLE
Documentation rewrite

### PR DESCRIPTION
**Pull request by mkevenaar**
Work in progress. Do not merge yet

- [x] 0_installing
  - [x] index.md (new) 
  - [x] docker_compose.md
  - [x] linux_service.md
  - [x] monsterpi.md
  - [x] windows_service.md
- [x] 1_developer_setup
  - [x] index.md
  - [x] setup_client.md
  - [x] setup_server.md
  - [x] setup_octofarm.md
- [ ] 2_configuration
  - [ ] settings
    - [ ] grid_settings.md
    - [ ] index.md
  - [ ] index.md
  - [ ] preconfiguration.md
  - [ ] updating_client_bundle.md
  - [ ] yaml_import_export.md
- [ ] 3_software_usage
  - [ ] creating_printers.md
  - [ ] index.md
  - [ ] organizing_floors.md
- [ ] 4_troubleshooting
  - [ ] index.md
- [x] index.md

### CODE REVIEW

**Review by davidzwa — APPROVED** [review]
So far I fully approve. The `nav_order` and `parent` tags of course we spoke about. 
I like the details in this PR.

Ill gladly review again later

File: docs/2_configuration/preconfiguration.md [hunk 1]
```diff
@@ -1,6 +1,13 @@
-# Pre-configuring the FDM Monster Server
-
-> :warning: **After each environment change the FDM Monster server must be restarted for the change to take effect!**
+---
+layout: default
+title: Pre-configuring the FDM Monster Server
+parent: Configuration
+nav_order: 1
```
Comment (davidzwa) [inline docs/2_configuration/preconfiguration.md:5]: `sidebar_position` does exist through the docs plugin that we have by default

Comment (mkevenaar) [reply docs/2_configuration/preconfiguration.md:5]: Todo when working on section 2 and above will be handled.

**Review by davidzwa — APPROVED** [review]
Current state is a massive improvement. Another approval 😄

**Review by davidzwa — APPROVED** [review]
LGTM! Eager to see this live, awesome job!!!

**Comment by davidzwa** [comment]
@all-contributors please add @mkevenaar for doc

**Pull request merged**